styleguides/Javascript/core.md

```diff
@@ -0,0 +1,3423 @@
+
+  ```javascript
+  // bad
+  !isRunning && startRunning();
+  
+  // good
+  if (!isRunning) {
+    startRunning();
+  }
+  ```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Comments
+
+  <a name="comments--multiline"></a><a name="17.1"></a>
+
+- [18.1](#comments--multiline) Use `/** ... */` for multi-line comments.
+
+  ```javascript
+  // bad
+  // make() returns a new element
+  // based on the passed in tag name
+  //
+  // @param {String} tag
+  // @return {Element} element
+  function make(tag) {
+  
+    // ...
+  
+    return element;
+  }
+  
+  // good
+  /**
+   * make() returns a new element
+   * based on the passed-in tag name
+   */
+  function make(tag) {
+  
+    // ...
+  
+    return element;
+  }
+  ```
+
+  <a name="comments--singleline"></a><a name="17.2"></a>
+- [18.2](#comments--singleline) Use `//` for single line comments. Place single line comments on a newline above the subject of the comment. Put an empty line before the comment unless it’s on the first line of a block.
+
+  ```javascript
+  // bad
+  const active = true;  // is current tab
+  
+  // good
+  // is current tab
+  const active = true;
+  
+  // bad
+  function getType() {
+    console.log('fetching type...');
+    // set the default type to 'no type'
+    const type = this.type || 'no type';
+  
+    return type;
+  }
+  
+  // good
+  function getType() {
+    console.log('fetching type...');
+  
+    // set the default type to 'no type'
+    const type = this.type || 'no type';
+  
+    return type;
+  }
+  
+  // also good
+  function getType() {
+    // set the default type to 'no type'
+    const type = this.type || 'no type';
+  
+    return type;
+  }
+  ```
+
+  <a name="comments--spaces"></a>
+- [18.3](#comments--spaces) Start all comments with a space to make it easier to read. eslint: [`spaced-comment`](https://eslint.org/docs/rules/spaced-comment)
+
+  ```javascript
+  // bad
+  //is current tab
+  const active = true;
+  
+  // good
+  // is current tab
+  const active = true;
+  
+  // bad
+  /**
+   *make() returns a new element
+   *based on the passed-in tag name
+   */
+  function make(tag) {
+  
+    // ...
+  
+    return element;
+  }
+  
+  // good
+  /**
+   * make() returns a new element
+   * based on the passed-in tag name
+   */
+  function make(tag) {
+  
+    // ...
+  
+    return element;
+  }
+  ```
+
+  <a name="comments--actionitems"></a><a name="17.3"></a>
+- [18.4](#comments--actionitems) Prefixing your comments with `FIXME` or `TODO` helps other developers quickly understand if you’re pointing out a problem that needs to be revisited, or if you’re suggesting a solution to the problem that needs to be implemented. These are different than regular comments because they are actionable. The actions are `FIXME: -- need to figure this out` or `TODO: -- need to implement`.<a name="comments--fixme"></a><a name="17.4"></a>
+- [18.5](#comments-
```

Comment:
I think we need a discussion on whether to use 2 spaces or 4 spaces for JS. There is no set standard on how many spaces you should use or whether to use spaces or tabs and it is purely subjective. Personally, I am of the preference of 4 spaces.